### PR TITLE
Komodo: cmake config changes

### PR DESCRIPTION
**Pull request by joakim-hove**
Internally in equinor we have a system for installing inhouse developed software called Komodo[1]. That system stresses the cmake features of both the packages and the operators to the limit.

Currently I am struggling to build [sunbeam](https://github.com/Statoil/sunbeam/blob/master/CMakeLists.txt) which depends on opm-common through:
```
find_package(opm-common REQUIRED)
```

My installed version of `opm-common-config.cmake` ends like this:
```cmake

  # extra code from variable OPM_PROJECT_EXTRA_CODE
  #ENABLE_ECL_INPUT is needed by opm-common-prereq.cmake
                                       set(ENABLE_ECL_INPUT ON)
                                       set(OPM_MACROS_ROOT /project/res/komodo/opm-common/root/share/opm)
                                       list(APPEND CMAKE_MODULE_PATH ${OPM_MACROS_ROOT}/cmake/Modules)
                                       include(OpmPackage) #Make macros availabe after find_package(opm-common)
                                        set(COMPARE_SUMMARY_COMMAND /project/res/komodo/opm-common/root/bin/compareSummary)
                                        set(COMPARE_ECL_COMMAND /project/res/komodo/opm-common/root/bin/compareECL)
                                            set(HAVE_ECL_INPUT 1)
                                            set(HAVE_ECL_OUTPUT 1)
  # end extra code

  # This call is at the bottom as we need to include
  # the OpmPackage Macros.
  set(CMAKE_MODULE_PATH "${CMAKE_MODULE_PATH}" /var/jenkins/jenkins-be/workspace/komodo-advanced/opm-common/project/res/komodo/opm-common/root/share/opm/cmake/Modules)
  include(opm-common-prereqs)
```

and when invoked from my the sunbeam build system this fails on the `include(OpmPackage)` with `"No such file or directory"`. Moving the explicit manipulation of the `CMAKE_MODULE_PATH` up seems to fix it?!

[1]: NB: `flow` is **not** distributed using komodo, so this not relevant for the `flow` hot-path.

### CODE REVIEW

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
@akva2: Does this look correct?

**Comment by joakim-hove** [comment]
jenkins build this please

File: cmake/Templates/opm-project-config.cmake.in [hunk 1]
```diff
@@ -29,11 +29,34 @@ if(NOT @opm-project_NAME@_FOUND)
...
   set (@opm-project_NAME@_LIBRARIES ${@opm-project_NAME@_LIBRARY} "@opm-project_LIBRARIES@")
+
+  # The purpose of this string replacement operation is to enable use of the
+  # generated opm-project-config.cmake file also in the situation where 'make
+  # install' has been invoked with the DESTDIR option:
+  #
+  # opm-common/build> cmake .. -DCMAKE_INSTALL_PREFIX=/real/prefix
+  # opm-common/budil> make install DESTDIR=/tmp/prefix
+  #
+  # downstream/build> cmake .. -DEST_PREFIX=/tmp/prefix -DCMAKE_PREFIX_PATH=/tmp/prefix
```
Comment (akva2) [inline cmake/Templates/opm-project-config.cmake.in]: -DDEST_PREFIX

Comment (joakim-hove) [reply cmake/Templates/opm-project-config.cmake.in]: Fixed and pushed - thank you.

**Comment by akva2** [comment]
it's slighly worrysome that we lose the extra code, but if those using this is not in need of that we can address that when the need surfaces. seems fine to me, and should not harm elsewhere. merging.

**Pull request merged**